xdsl/printer.py

```diff
@@ -534,10 +535,25 @@ def print_attribute(self, attribute: Attribute) -> None:
 
         if isinstance(attribute, FloatAttr):
             value = attribute.value
-            attr_type = cast(
-                FloatAttr[Float16Type | Float32Type | Float64Type], attribute
-            ).type
-            self.print_string(f"{value.data:.6e} : ")
+            attr_type = cast(AnyFloatAttr, attribute).type
+            if np.isnan(value.data) or np.isinf(value.data):
+                if isinstance(attr_type, Float16Type):
+                    self.print_string(
+                        f"{hex(np.float16(value.data).view('uint16'))} : "
+                    )
+                elif isinstance(attr_type, Float32Type):
+                    self.print_string(
+                        f"{hex(np.float32(value.data).view('uint32'))} : "
+                    )
+                elif isinstance(attr_type, Float64Type):
+                    self.print_string(
+                        f"{hex(np.float64(value.data).view('uint64'))} : "
+                    )
+                else:
+                    # todo
```

Comment:
what is the todo?

Reply:
To add support for other types such as f80, f128

Reply:
Updated this in the code.